libdirections-onboard/src/test/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouterTest.kt

```diff
@@ -1,24 +1,391 @@
 package com.mapbox.navigation.route.onboard
 
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.extensions.applyDefaultParams
+import com.mapbox.navigation.base.extensions.coordinates
+import com.mapbox.navigation.base.logger.Logger
+import com.mapbox.navigation.base.logger.model.Message
+import com.mapbox.navigation.base.logger.model.Tag
+import com.mapbox.navigation.base.route.Router
 import com.mapbox.navigation.navigator.MapboxNativeNavigator
 import com.mapbox.navigation.route.onboard.model.Config
+import com.mapbox.navigation.utils.exceptions.NavigationException
+import com.mapbox.navigation.utils.tests.MainCoroutineRule
+import com.mapbox.navigation.utils.tests.runBlockingTest
+import com.mapbox.navigator.RouterResult
+import io.mockk.every
 import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import junit.framework.Assert.assertEquals
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
 import org.junit.Assert
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
 
+@InternalCoroutinesApi
+@ExperimentalCoroutinesApi
+@RunWith(RobolectricTestRunner::class)
 class MapboxOnboardRouterTest {
+
+    @get:Rule
+    var coroutineRule = MainCoroutineRule()
+
     private lateinit var onboardRouter: MapboxOnboardRouter
-    private val navigator: MapboxNativeNavigator = mockk()
-    private val tilePath = "tiles"
+
+    private val navigator: MapboxNativeNavigator = mockk(relaxed = true)
```

Comment:
NIT: preferably we use only `relaxUnitFun`. Same for all below.

Reply:
fixed